### ISSUE TITLE
Problems detecting network/Internet connextion when using dnscrypt-proxy

### ISSUE THREAD

**Issue by tuqueque** [GitSquared/edex-ui#749]
Hi, I admit I'm not a frequent eDEX-UI user, but recently I tried it and the "Network Status" panel shows "offline". Under the IPv4 tag there's "--.--.--.--" and under "Ping", there's "--ms".

I read issue #460 ( https://github.com/GitSquared/edex-ui/issues/460 ) and most things look the same, but that issue seems to be solved while I'm still having connection report issues in my case. I'm using version 2.2.2, which is the latest version at the time of writing this bug report... I'm running Manjaro Linux, Kernel: x86_64 Linux 5.6.16-1.

I think my issue is related to dnscrypt-proxy running on my system ( https://github.com/DNSCrypt/dnscrypt-proxy ), I already had to change eDEX-UI to use port 3001 instead of 3000 because dnscrypt-proxy uses it for some fancy setups that I have set up ( https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Local-DoH )

I also took the liberty of running the diagnostics commands suggested in issue #460, here are the results:

from: https://github.com/GitSquared/edex-ui/issues/460#issuecomment-473678409
```
window.si.networkInterfaces(d => console.log(JSON.stringify(d, 0, " ")))
Promise {<pending>}
__proto__: Promise
[[PromiseStatus]]: "resolved"
[[PromiseValue]]: Array(2)
VM279:1 [
 {
  "carrierChanges": 0,
  "duplex": "",
  "iface": "lo",
  "ifaceName": "lo",
  "internal": true,
  "ip4": "127.0.0.1",
  "ip6": "::1",
  "mac": "",
  "mtu": 65536,
  "operstate": "unknown",
  "speed": -1,
  "type": "virtual"
 },
 {
  "carrierChanges": 8,
  "duplex": "",
  "iface": "wlp3s2",
  "ifaceName": "wlp3s2",
  "internal": false,
  "ip4": "192.168.43.252",
  "ip6": "fe80::a380:4521:ada0:ebcc",
  "mac": "00:18:e7:28:44:f0",
  "mtu": 1500,
  "operstate": "up",
  "speed": -1,
  "type": "wireless"
 }
]
```
From: https://github.com/GitSquared/edex-ui/issues/460#issuecomment-473682838
```
output = {iface: window.mods.netstat.iface, internalIPv4: window.mods.netstat.internalIPv4, externalIPinfo: window.mods.netstat.ipinfo, lastReachFinished: window.mods.netstat.lastconn.finished, lastReachHeader: window.mods.netstat.lastconn._header, offlineVerdict: (window.mods.netstat.offline) ? "offline" : "online"};console.log(JSON.stringify(output, 0, " "));
VM282:1 {
 "iface": "wlp3s2",
 "internalIPv4": "192.168.43.252",
 "lastReachFinished": true,
 "lastReachHeader": "GET /json/ HTTP/1.1\r\nHost: freegeoip.app\r\nConnection: keep-alive\r\n\r\n",
 "offlineVerdict": "offline"
}
undefined
```
Hopefully this will help.
Greets!

**Comment by GitSquared** [comment]
Great job digging through the issues and trying debugging stuff. :ok_hand:

The software pings an IPv4 address to check external connectivity, by default its set to use Cloudflare's DNS at `1.1.1.1` (we need a default, right?).

Check that you have external IPv4 connectivity, and whether dnscrypt blocks the connection to 1.1.1.1. If it does, you can change the address used by edex in the settings - see the wiki on this github repo.

I actually use dnscrypt-proxy myself, so you should be able to make it work.

**Comment by tuqueque** [comment]
Solved!... I just changed the ping's IP to 1.0.0.1 and not it shows "online" and it gives me a ping reading. I still don't see IPv4 address, but I don't really care much about that, I'm just curious about why is not showing... I also tried to ping to Google's 8.8.8.8 and to Quad9's 9.9.9.9 but the only one that works is 1.0.0.1 (I'm using 1.1.1.1 as the main DNS service in dnscrypt-proxy, maybe there's a conflict there, I have no idea).

Anyway, thanks for pointing me in the right direction. At least now I'm able to see myself "online".
